### ISSUE TITLE
OPENSTREAM broken in lisp.sysout for {unix} FDEV - \GENERIC.READCCODE is not an external format

### ISSUE THREAD

**Issue by nbriggs** [Interlisp/medley#457]
Trying to run a lisp.sysout, built 3-Sep-2021:
```
(OPENSTREAM "{unix}/tmp/output.txt" 'OUTPUT 'NEW)
\GENERIC.READCCODE is not an external format
```
but it works fine for `"{dsk}/tmp/output.txt"`

**Comment by nbriggs** [comment]
`\DO.PARAMS.AT.OPEN` was called with an output stream on the `{UNIX}` device, OUTPUT, and NIL for the parameters... which resulted in a call to
```
(\EXTERNALFORMAT the-output-stream '\GENERIC.READCCODE)
```

**Comment by nbriggs** [comment]
It's failing because the DEFAULTEXTERNALFORMAT of the UNIX FDev is indeed the atom \GENERIC.READCCODE

**Comment by nbriggs** [comment]
There's another problem, though it's not the cause of this problem -- the FDEV struct in the Maiko C code no longer matches the Lisp FDEV object.

**Comment by nbriggs** [comment]
`\UFS.CREATE.DEVICE` creates an FDEV with wrong info.  Perhaps it was not recompiled with the correct FDEV datatype definition?

**Comment by rmkaplan** [comment]
Yes, I didn't have the full.database at the time that I made this switch, so didn't know what needed to be recompiled.  And then I forgot that I needed to do that. There are a bunch of other FDEV creators that I will recompile.

I haven't seen this because I typically run in the full.sysout which has Unicode loaded, and Unicode says that the default format for UNIX is :UTF-8, so \EXTERNALFORMAT doesn't fall through to get the default format from the device.  But DSK has :XCCS even in Lisp.

\GENERIC.READDCODE no longer exists, deprecated in favor of the even more generic \INCCODE.

My understanding of the maiko/lisp compatibility for streams, fdevs and other datatypes that are known at both levels is that old fields and field descriptors cannot be changed unilaterally, but that fields can be added.  Is that not the case?

**Comment by nbriggs** [comment]
Perfect, thanks.  I recompiled UFS and found it worked but hadn't got to saying so.
Re the structure matching -- it depends on what it's used for.  I *think* that FDEVs are only known for debugging the C code, so it doesn't render the system non-functional if they don't change at the exact same time.  For STREAMs, the BIN opcode looks at the BINABLE, COFFSET, and CBUFPTR fields so they need to be in sync, and the type number of the object must be 11 (or, if it must change, it needs to change to a new reproducible number) .

**Comment by nbriggs** [comment]
Closed by PR #463
